File: brickport-services/src/queries/game-queries.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace BrickPort.Services.Queries
{
    public class GameSummary
    {
        public string Id { get; set; }
        public DateTime DateUtc { get; set; }
        public string Winner { get; set; }
        public PlayerScoreSummary[] PlayerScores { get; set; }
    }

    public class PlayerScoreSummary 
    {
        public string PlayerId { get; set; }
        public string PlayerName { get; set; }
        public string Color { get; set; }
        public int VictoryPoints { get; set; }
    }

    public interface IGameQueries
    {
        Task<IEnumerable<GameSummary>> SummaryAsync();
        Task<GameSummary> SummaryAsync(string id);
        Task<IEnumerable<GameSummary>> SummaryByDateAsync(DateTime? startUtc, DateTime? endUtc);
    }
}

File: brickport-ui/src/brickport-services/queries/game-queries.ts
export interface IGameSummary {
    id: string | null;
    dateUtc: string;
    winner: string | null;
    inProgress: boolean;
    playerScores: IPlayerScoreSummary[];
}

export interface IPlayerScoreSummary {
    playerId: string | null;
    playerName: string | null;
    color: string | null;
    victoryPoints: number;
}

export class GameSummary implements IGameSummary {
    id!: string | null;
    dateUtc!: string;
    winner!: string | null;
    inProgress!: boolean;
    playerScores!: IPlayerScoreSummary[];
    constructor(state?: IGameSummary) {
        if (state) {
            Object.assign(this, state);
        }
        this.id = this.id || null;
        this.dateUtc = this.dateUtc || new Date().toISOString();
        this.winner = this.winner || null;
        this.inProgress = this.inProgress || false;
        this.playerScores = this.playerScores || [];
    }
}

export interface IGameQueries {
    summaryAsync(): Promise<IGameSummary[]>;
    summaryAsync(id: string): Promise<IGameSummary>;
    summaryByDateAsync(startUtc: string | null, endUtc: string | null): Promise<IGameSummary[]>;
}